File: src/MaksIT.Models/LetsEncryptServer/ApiKeys/Search/SearchApiKeyEntityScopeResponse.cs
using MaksIT.Core.Abstractions.Webapi;

namespace MaksIT.Models.LetsEncryptServer.ApiKeys.Search;

/// <summary>One API key entity scope row in search results (includes parent API key description).</summary>
public class SearchApiKeyEntityScopeResponse : ResponseModelBase {
  public required Guid Id { get; set; }
  public required Guid ApiKeyId { get; set; }
  public string? Description { get; set; }
  public required Guid EntityId { get; set; }
  public string? EntityName { get; set; }
  /// <summary>Numeric discriminator / enum value (product-specific).</summary>
  public int EntityType { get; set; }
  /// <summary>Permission flags as integer (product-specific).</summary>
  public int Scope { get; set; }
}

File: src/MaksIT.WebUI/src/models/apiKey/search/SearchApiKeyEntityScopeResponse.ts
import type { ResponseModelBase } from '@maks-it.com/webui-contracts'
import { ScopeEntityType } from '../../ScopeEntityType'
import { ScopePermission } from '@maks-it.com/webui-core'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.Search.SearchApiKeyEntityScopeResponse</c>. */
export interface SearchApiKeyEntityScopeResponse extends ResponseModelBase {
  id: string
  apiKeyId: string
  description?: string | null
  entityId: string
  entityName?: string | null
  entityType: ScopeEntityType
  scope: ScopePermission
}
